src/pages/slots/CreateSlot.tsx

```diff
@@ -0,0 +1,63 @@
+import { MainContainer, Label, Input, Button, Select, InputContainer, EditContainer, EditCapacity, TitleContainer, Title, ScreenContainer } from "./CreateSlot.styles";
+import EditIcon from "../../assets/edit-icon.png";
+import EditSlotForm from "./forms/EditCapacityForm";
+import { useRef, useState } from "react";
+import type { ReactNode } from "react";
+import AddIcon from "../../assets/add-icon.svg";
+import Modal from "../../components/modal/Modal";
+import BackIcon from "../../assets/back-icon.png";
+import { useNavigate } from "react-router-dom";
+
+function CreateSlot() {
+
+
+
+    const DaysOfWeek = ["Lunes", "Martes", "Miércoles", "Jueves", "Viernes", "Sábado", "Domingo"];
+
+    const [modal, setModal] = useState<ReactNode | null>(null);
+
+    const editCapacityRef = useRef<any>(null);
+
+    const navigate = useNavigate();
+
+    return (
+        <MainContainer>
+            <TitleContainer>
+                <Title>
+                    <img
+                        src={BackIcon}
+                        alt="back-icon"
+                        onClick={() => navigate(-1)}
+                        style={{ cursor: "pointer" }}
+                    ></img>
+                    MIS TURNOS
+                </Title>
+            </TitleContainer>
+            <ScreenContainer>
+                <InputContainer>
+                    <EditContainer>
+                        <Label >Cupos disponibles</Label>
+                        <EditCapacity onClick={() => setModal(<EditSlotForm />)}>Editar<img src={EditIcon} width={16} height={16}></img></EditCapacity>
+                    </EditContainer>
+                    {modal && (
+                        <Modal onClose={() => setModal(false)} content={<EditSlotForm ref={editCapacityRef} />} showButtons={true} contentRef={editCapacityRef} onConfirm={() => {
+                            // Vacío - el Modal ya llama a submitForm automáticamente
+                        }} />
```

Comment:
Siguiendo el comentario anterior, es aca donde nosotros en el onConfirm deberiamos llamar al metodo submit del ref
  
```suggestion
                        <Modal onClose={() => setModal(false)} content={<EditSlotForm ref={editCapacityRef} />} showButtons={true} contentRef={editCapacityRef} onConfirm={handleConfirmModal} />
```

siendo el handleConfirmModal este metodo
```
    const handleConfirmModal = async () => {
        const response = await editCapacityRef.current.submitForm();
        if(response) {
            setModal(false);
        }
    }
```

de esta manera abstraemos al modal de lo que se quiere hacer en el confirmar, porque solo vamos a usar el ref en este caso para crear el slot, logrando que nuestro componente esté más limpio. Ahora, fijate que el `submitForm()` esta devolviendo una respuesta, esto lo hice para que solo se cierre el modal y la ejecucion del metodo del formulario se completa correctamente (esto puede variar segun la experiencia deseada), dejo en otro comentario como cambia el `submitForm()` en el formulario